web/packages/teleport/src/components/AgentButtonAdd/AgentButtonAdd.tsx

```diff
@@ -26,6 +26,9 @@ export default function AgentButtonAdd(props: Props) {
   const { canCreate, isLeafCluster, onClick, agent, beginsWithVowel } = props;
   const disabled = isLeafCluster || !canCreate;
 
+  // Don't render button if feature hiding is enabled.
+  const hidden = (isLeafCluster || !canCreate) && cfg.hideInaccessibleFeatures;
```

Comment:
Couldn't we use `disabled` instead of repeating `(isLeafCluster || !canCreate)`?